torch/fx/immutable_collections.py

```diff
@@ -26,3 +29,22 @@ def _create_immutable_container(base, mutable_functions):
 immutable_dict = _create_immutable_container(dict, ['__delitem__', '__setitem__', 'clear', 'pop', 'popitem', 'update'])
 immutable_dict.__reduce__ = lambda self: (immutable_dict, (iter(self.items()),))
 compatibility(is_backward_compatible=True)(immutable_dict)
+
+
+# Register immutable collections for PyTree operations
```

Comment:
These should go in torch/utils/_pytree, right?
- It would be good to have all the definitions near each other (https://github.com/pytorch/pytorch/blob/master/torch/utils/_pytree.py#L65)
- I don't know if this file gets imported when someone uses torch.utils._pytree or if it only gets imported when someone calls `import torch.fx`. We want general users of pytorch's PyTree to be able to benefit from these, not just when they `import torch.fx`.

Reply:
Oh, nevermind, immutable list/dict aren't python constructs, they're defined here. Please ignore.